field/bitmap.go

```diff
@@ -19,7 +19,7 @@ type Bitmap struct {
 	bitmap *utils.Bitmap
 }
 
-func NewBitmap(spec *Spec) Field {
+func NewBitmap(spec *Spec) *Bitmap {
```

Comment:
I added a paragraph at the end of the PR description explaining it. Let me know what you think:

> Finally, we have updated all of the constructors in the field package to return a pointer to their concrete types rather than the field.Field interface. This change is inspired by the 'Accept Interfaces Return Struct' Golang principle. Interfaces are implemented implicitly in Go and because our new Subfields type implements both the field.Field and field.Struct interfaces, returning its concrete type allows us to leverage methods from both interfaces without having to type cast from one interface to the other.